server/src/main/java/org/opensearch/extensions/RegisterRestActionsResponse.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
```

Comment:
Do we need ES license if this is a new file?

Reply:
See comment above to @saratvemulapalli .

Technically as it's a new file it could have its own license header. But the license check/formatting for OpenSearch project where this will go is not set up for a mixture of license headers.   So if we change it, it will break current build automation license checks.   Do we need to open another issue in this project to discuss this issue? Sounds like it.  I think this needs a much higher level / legal decision.

Reply:
Sure. We can open an issue to move the discussion there

Reply:
Actually it's answered here.  Should have a new license. Will fix.  https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md#license-headers